README.md

```diff
@@ -1,135 +1,450 @@
-increasing.
+### Changing the default network
 
-Explore available commands for managing and monitoring the network:
+If you prefer to keep the `hardhat` network as the default network for your Hardhat project, you can configure `kadena_devnet0` as a separate network.
 
-```sh
-./network help
-```
+To change the default network:
 
-When done shut down the network with:
+1. Open the `hardhat.config.js` file for your Hardhat project in your code editor. 
 
-```sh
-./network devnet stop
-```
+2. Remove or comment out the following line from the network settings:
 
-If, for some reason, the network gets stuck the chainweb-node can be restarted
-via:
+   ```sh
+   defaultNetwork: "kadena_devnet0",
+   ```
 
-```sh
-./network devnet restart
+1. Run hardhat command using the --network kadena_devnet0 command-line option.
+   
+   For example, run unit tests on the Chainweb EVM development network like this:
+
+   ```sh
+   npx hardhat test --network kadena_devnet0
+   ```
+   
+   Similarly, you can run scripts on the Chainweb EVM development network like this:
+
+   ```sh
+   npx hardhat run scripts/deploy.js --network kadena_devnet0
+   ```
+
+   You can then run the same commands without the `--network` command-line option to execute them on the default Hardhat network instance. 
+   If your smart contract uses any Kadena-specific precompile functions or other Kadena-specific or Chainweb-specific features, you might be unable to run the contract on the Hardhat network.
+   If you have a smar contract that requires Hardhat-specific features, you might be unable to run the contract on the Kadena development network.
+
+## Signing transactions and switching chains
+ 
+By default, when you call a Solidity smart contract from a test file or a script using `ethers`, the transaction is executed by the first signer address and private key for the local development environment. 
+This is the true whether the local development environment is the default Hardhat network or the Kadena development network. 
+For example, the following `transferCrossChain` transaction is signed by the account that corresponds to the first address in the list of addresses for the current network context:
+
+```javascript
+const tx = await token0.transferCrossChain(receiver.address, amount, token1Info.chain);
 ```
 
-### Test Simple Token Contract
+This address is the `msg.sender` for the transaction. 
 
-Before running the simple token contract tests for the first time one needs to
-install hardhat. This can be done with the following command.
+In the test files in the `solidity/test` directory, this address is to the **deploying signer** you see displayed when you execute the tests: 
 
 ```sh
-./network solidity setup
+Found 2 Kadena devnet networks: kadena_devnet0, kadena_devnet1
+Deploying with signer: 0x8849BAbdDcfC1327Ad199877861B577cEBd8A7b6 on network kadena_devnet0
+Deploying with signer: 0x8849BAbdDcfC1327Ad199877861B577cEBd8A7b6 on network kadena_devnet1
 ```
```

Comment:
I changed the text that is printed to:
Deployed to address 0x0Be392280A18d06DDC9CdD162485CB3334abf8e7 with signer: 0x8849BAbdDcfC1327Ad199877861B577cEBd8A7b6 on network kadena_devnet0
Deployed to address 0x03C92E75C2e211E9A57B3f31e7354E9D0aF6F257 with signer: 0x8849BAbdDcfC1327Ad199877861B577cEBd8A7b6 on network kadena_devnet1

The addresses change each time, but I think it's OK to hardcode them in this example text.